install.c

```diff
@@ -170,18 +179,27 @@ int inst_menu_cb(dia_item_t di)
  */
 int inst_choose_netsource()
 {
+  int i, j;
+  slist_t *sl;
   dia_item_t di;
-  dia_item_t items[] = {
+  dia_item_t items[MAX_DIALOG_ITEMS] = {
     di_netsource_ftp,
     di_netsource_http,
     di_netsource_https,
     di_netsource_nfs,
     di_netsource_smb,
     di_netsource_tftp,
-    di_none
   };
 
-  if(!(config.test || config.net.cifs.binary)) items[3] = di_skip;
+ j = 6;        // first 6 items have aleady been filled in
```

Comment:
Maybe try to avoid magic numbers in the middle of the code? Add some #define ? Or maybe use a sizeof() of a static array?